### ISSUE TITLE
Dialog is sometimes ugly

### ISSUE THREAD

**Issue by MCGallaspy** [certbot/certbot#2025]
I ran the command `./letsencrypt-auto certonly -d foo.com --test-cert` on Ubuntu 14.04.3 and saw the following weird prompt:

![ss3](https://cloud.githubusercontent.com/assets/8888020/12005187/688e8410-ab4b-11e5-94d9-d74f6ace508f.png)

The borders are misaligned/missing, and use unexpected characters.

**Comment by pde** [comment]
Dialog seems to produce messed up output on some terminals.  We haven't been able to work out why, or what to do about it.

**Comment by MCGallaspy** [comment]
I did a cursory poking around, seems like it might be a bug in `zope.component`? I'll take a closer look as time permits.

**Comment by pde** [comment]
That would be surprising.  There's a `zope.interface` wrapper around all the `dialog` calls (in order to support the `--text` command line argument, for instance) but I'd be surprised if `zope.component` was modifying the calls out to the python `dialog` module.

**Comment by pde** [comment]
That interface is documented [here](https://letsencrypt.readthedocs.org/en/latest/api/interfaces.html#letsencrypt.interfaces.IDisplay).  The dialog / ncurses implementation is [here](https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt/display/util.py#L25)

**Comment by jpluimers** [comment]
There is a similar `yast` issue on opensuse (https://wiert.me/2016/06/09/tumbleweed-local-console-yast-linedrawing-characters-garbage-after-first-reboot/) for which the workaround is to run `/bin/unicode_start` once by any user.

**Comment by bmw** [comment]
Due to problems like this (as well as other motivating factors), we've completely removed `dialog` as a dependency of this project in favor of our text interface for our next release. With this in mind, I'm closing this issue.

**Comment by jpluimers** [comment]
Please post a reference to that new text interface in case other people need to find it after upgrading.

**Comment by pde** [comment]
@jpluimers not clear what you're asking for there, but essentially Certbot 0.10.0 is going to behave a lot like `certbot -t` did for version 0.9.3 and below.

**Comment by jpluimers** [comment]
Ah. I never noticed there is `certbot -t` as it's not in the `certbot --help` list.

I actually like the `certbot -t` UI. Maybe that is because I'm from the Infocom era (:

**Comment by pde** [comment]
I'm pretty sure `-t` was [in the help output](https://github.com/certbot/certbot/blob/ce4e00569e6d8ed3d51c5a078d4281bec5f8e5f0/docs/cli-help.txt#L25) for 0.9.3 and below.

**Comment by jpluimers** [comment]
I just discovered I was still at the master branch on commit https://github.com/certbot/certbot/commit/da3332ccfaad08bd9dece68f5ae71308acb1d776 which displays itself as 0.9.3 but already has that part from the help removed.

I'm not enough of a git and github person to show that in a commit graph.

I just now did a , but since then I've done `git fetch` and `git pull` and it's the same: it shows version `letsencrypt 0.9.3` but doesn't show the `-t` portion of the help:

```
# git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
# git rev-parse HEAD
da3332ccfaad08bd9dece68f5ae71308acb1d776
# ./certbot-auto --version
letsencrypt 0.9.3
# ./certbot-auto --help

  certbot-auto [SUBCOMMAND] [options] [-d domain] [-d domain] ...

Certbot can obtain and install HTTPS/TLS/SSL certificates.  By default,
it will attempt to use a webserver both for obtaining and installing the
cert. Major SUBCOMMANDS are:

  (default) run        Obtain & install a cert in your current webserver
  certonly             Obtain cert, but do not install it (aka "auth")
  install              Install a previously obtained cert in a server
  renew                Renew previously obtained certs that are near expiry
  revoke               Revoke a previously obtained certificate
  register             Perform tasks related to registering with the CA
  rollback             Rollback server configuration changes made during install
  config_changes       Show changes made to server config during installation
  plugins              Display information about installed plugins

Choice of server plugins for obtaining and installing cert:

  --apache          Use the Apache plugin for authentication & installation
  --standalone      Run a standalone webserver for authentication
  --nginx           Use the Nginx plugin for authentication & installation
  --webroot         Place files in a server's webroot folder for authentication

OR use different plugins to obtain (authenticate) the cert and then install it:

  --authenticator standalone --installer apache

More detailed help:

  -h, --help [topic]    print this message, or detailed help on a topic;
                        the available topics are:

   all, automation, paths, security, testing, or any of the subcommands or
   plugins (certonly, renew, install, register, nginx, apache, standalone,
   webroot, etc.)
```

After fetch and pull:

```
# git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
# git rev-parse HEAD
839ff7a2657e6854d053573268b0260f6842e2bd
θ100° [root@revue:~/Versioned/certbot] # ./certbot-auto --version
letsencrypt 0.9.3
θ100° [root@revue:~/Versioned/certbot] 2s # ./certbot-auto --help

  certbot-auto [SUBCOMMAND] [options] [-d domain] [-d domain] ...

Certbot can obtain and install HTTPS/TLS/SSL certificates.  By default,
it will attempt to use a webserver both for obtaining and installing the
cert. Major SUBCOMMANDS are:

  (default) run        Obtain & install a cert in your current webserver
  certonly             Obtain cert, but do not install it (aka "auth")
  install              Install a previously obtained cert in a server
  renew                Renew previously obtained certs that are near expiry
  revoke               Revoke a previously obtained certificate
  register             Perform tasks related to registering with the CA
  rollback             Rollback server configuration changes made during install
  config_changes       Show changes made to server config during installation
  plugins              Display information about installed plugins

Choice of server plugins for obtaining and installing cert:

  --apache          Use the Apache plugin for authentication & installation
  --standalone      Run a standalone webserver for authentication
  --nginx           Use the Nginx plugin for authentication & installation
  --webroot         Place files in a server's webroot folder for authentication

OR use different plugins to obtain (authenticate) the cert and then install it:

  --authenticator standalone --installer apache

More detailed help:

  -h, --help [topic]    print this message, or detailed help on a topic;
                        the available topics are:

   all, automation, paths, security, testing, or any of the subcommands or
   plugins (certonly, renew, install, register, nginx, apache, standalone,
   webroot, etc.)
```

**Comment by pde** [comment]
* nods. Unfortunately you need to do `certbot --help all`.  We have too many flags to show them all by default. But also, we're doing a better job of this in 0.10.0 since we landed #3883 .

**Comment by jpluimers** [comment]
Ah. Now I see. Which means that ideally`certbot --help all` is one of the things that really should get displayed when performing a plan `certbot --help` (:
